lib/manifolds/project/project.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Manifolds
+  module API
+    # Projects API
+    class Project
+      attr_reader :name, :directory
+
+      def initialize(name, directory: Pathname.new(File.join(Dir.pwd, name)))
+        self.name = name
+        self.directory = directory
+      end
+
+      def init
```

Comment:
As a general rule, things like this should be a class method rather than an instance method.
```ruby
def initialise(name, directory:)
  @name = name
  @directory = Pathname(directory)
end
# ...
def self.create(name, directory: Pathname.pwd.join(name))
  new(name, directory: directory).tap do |project|
    [project.workspaces_directory, project.vectors_directory].each(&:mkpath)
  end
end
```
That way instead of this imperativeness:
```ruby
project = Manifolds::API::Project.new("some name")
project.init 
# etc
```
You have this:
```ruby
project = Manifolds::API::Project.create("some name")
# etc
```

Reply:
And done: https://github.com/bustle/manifolds/pull/9/commits/32ec4c0dedc7b11308430316b436c36bfce18d24